### DISCUSSION TITLE
HOMEBREW_BOOTSNAP on Apple Silicon?

### DISCUSSION THREAD

**Discussion by luckman212** [Homebrew/discussions#4629] [Everyday usage]
I was reading `man brew` and stumbled on an environment variable called `HOMEBREW_BOOTSNAP`, which says:

```
HOMEBREW_BOOTSNAP
    If set, use Bootsnap to speed up repeated brew calls. A no-op when using
    Homebrew's vendored, relocatable Ruby on macOS (as it doesn´t work).
```

Sounds like I would want this enabled! I don't quite understand what the possible drawbacks are of enabling it though. I did read (in the [3.0.0 announcement](https://brew.sh/2021/02/05/homebrew-3.0.0/)) that it doesn't yet work on AppleSilicon. Not sure if that's still the case as I write this, in July 2023. Is there any harm in having `export HOMEBREW_BOOTSNAP=1` in my `.bash_profile` ?

**Comment by luckman212** [comment]
Does anyone know the answer?

**Comment by Bo98** [comment]
It will generally make things a bit faster by maybe a quarter of a second or so. The drawback that means we can't enable this by default is `brew` will install this gem on startup when you enable it (so the initial run after you enable it will take a few seconds), and because it is a native gem this requires a functional compiler etc. This isn't necessarily guaranteed on every machine (e.g. https://github.com/Homebrew/brew/issues/10703 when we trialed enabling it by default for a few hours). 

Apple Silicon restriction you describe was lifted in [4.1.15](https://github.com/Homebrew/brew/releases/tag/4.1.15).

So in short: if you enable it and there's no errors - it should work well! As noted in https://github.com/orgs/Homebrew/discussions/4994, you probably don't want to `brew cleanup` too religiously with it on (e.g. don't do it after _every_ brew run), but you can still on regular occasions. The first run after a `brew cleanup` will be like it wasn't enabled for that one run.
